File: minify.js
// remove single line comments
function single_comments(text){
	var re = '/\/\/.*/g';
	return text.replace(re, '');
}

// removes multi-line comments
function multi_comments(text){
	var text = text.replace('\n', '');
	var re = '/(\/\*)(.*)(\*\/)/g';
	return text.replace(re, '');
}

// parser
function parser(text){
	text = single_comments(text);
	text = multi_comments(text);
	return text.split(/\s+/).join(' ');
}

File: minify.py
import re

# remove single line comments
def single_comments(text):
	return re.sub(r'//.*', '', text)

# removes multi-line comments
def multi_comments(text):
	return re.sub(r'(\/\*)(.*)(\*\/)', '', text.replace('\n', ''))

# parser
def parser(text):
	text = single_comments(text)
	text = multi_comments(text)
	return ' '.join(text.split())
